AfrikaBurn/MapViewController.swift

```diff
@@ -0,0 +1,194 @@
+    let overlayImage: UIImage
+    
+    init(overlay: MKOverlay, overlayImage: UIImage) {
+        self.overlayImage = overlayImage
+        super.init(overlay: overlay)
+    }
+    
+    override func draw(_ mapRect: MKMapRect, zoomScale: MKZoomScale, in context: CGContext) {
+        guard let imageRef = overlayImage.cgImage else {
+            return
+        }
+        let theMapRect = overlay.boundingMapRect
+        let theRect = rect(for: theMapRect)
+        
+        context.scaleBy(x: 1, y: -1)
+        context.translateBy(x: 0, y: -theRect.height)
+        context.draw(imageRef, in: theRect)
+    }
+}
+
+
+extension MapViewController: MKMapViewDelegate {
+    
+    
+    func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
+        // If annotation is not of type RestaurantAnnotation (MKUserLocation types for instance), return nil
+        if !(annotation is BurnAnnotation){
+            return nil
+        }
+        
+        var annotationView = self.mapView.dequeueReusableAnnotationView(withIdentifier: "Pin")
+        
+        if annotationView == nil{
+            annotationView = MKPinAnnotationView(annotation: annotation, reuseIdentifier: "Pin")
+            annotationView?.canShowCallout = true
+        }else{
+            annotationView?.annotation = annotation
+        }
+        
+        let burnAnnotation = annotation as! BurnAnnotation
+        
+        let blurb = UILabel(frame: CGRect(x: 0,y: 0,width: annotationView!.frame.size.width,height: 200))
+        blurb.font = UIFont(name: "Helvetica", size: 12)
+        blurb.text = burnAnnotation.element?.shortBlurb
+        blurb.numberOfLines = 6
+        annotationView?.detailCalloutAccessoryView = blurb
+        
+        
+        // Left Accessory
+        // We could make this change the icon based on the element type. Eg. mutant vehicles etc.
+        annotationView?.leftCalloutAccessoryView = UIImageView(image: UIImage(named: "fire.png"))
+        
+        // Right accessory view
+        
+        let button = UIButton(type: .detailDisclosure)
+        button.frame = CGRect(x: 0, y: 0, width: 30, height: 30)
+        
+        annotationView?.rightCalloutAccessoryView = button
+        return annotationView
+    }
+    
+    func mapView(_ mapView: MKMapView, annotationView view: MKAnnotationView, calloutAccessoryControlTapped control: UIControl) {
+        
+        let annotation = view.annotation
+        if !(annotation is BurnAnnotation){
+            return
+        }
+        
+        let burnAnnotation = annotation as! BurnAnnotation
+        let detail = BurnElementDetailViewController.create(camp: burnAnnotation.element!)
+        navigationController?.pushViewController(detail, animated: true)
+        
+    }
+    
+}
+
+
+class BurnAnnotation: NSObject, MKAnnotation {
+    
+    var coordinate: CLLocationCoordinate2D
+    var element: AfrikaBurnElement?
+    var title: String?
+    var image: UIImage?
```

Comment:
title has to be optional to conform with MKAnnotation. have made the rest required in the constructor 👍